open_optical_gating/cli/parameters.py

```diff
@@ -106,6 +113,8 @@ def update(
     Note: users should not use parameters.update(), i.e. a dictionary update
     as it will not automatically update certain keys.
     """
+    # TODO: JT writes: what is the *role* of this function? The “note” in the comment is not clear to me.
+    # Under what circumstances should this function used, by whom, and why? What is the alternative?
```

Comment:
So this is an old school thing that could probably solved by making settings a class. Basically, there are two items in 'parameters' that have a human-friendly and a computer-friendly version, i.e. referenceFrame (in frames) and targetSyncPhase (in rads). If you just update referenceFrames in the settings dictionary then the computer is still using targetSyncPhase - so settings should never be manually updated. But rather, should always be updated with this function.